test/test_parallel.jl

```diff
@@ -0,0 +1,58 @@
+# 2D LSRTM/FWI parallelization test with 2 sources and 2 vintages
+#
+# Ziyi Yin, ziyi.yin@gatech.edu
+# August 2021
+
+parsed_args = parse_commandline()
+
+nlayer = parsed_args["nlayer"]
+tti = parsed_args["tti"]
+fs =  parsed_args["fs"]
+
+### Model
+model, model0, dm = setup_model(tti, 4)
+q, srcGeometry, recGeometry, info = setup_geom(model; nsrc=4)
+q1 = q[[1,2]]
+q2 = q[[3,4]]
+srcGeometry1 = subsample(srcGeometry,[1,2])
+srcGeometry2 = subsample(srcGeometry,[3,4])
+recGeometry1 = subsample(recGeometry,[1,2])
+recGeometry2 = subsample(recGeometry,[3,4])
+info1 = subsample(info,[1,2])
+info2 = subsample(info,[3,4])
+
+dt = srcGeometry.dt[1]
+
+opt = Options(sum_padding=true, free_surface=fs)
+F1 = judiModeling(info1, model, srcGeometry1, recGeometry1; options=opt)
+F2 = judiModeling(info2, model, srcGeometry2, recGeometry2; options=opt)
+
+# Observed data
+dobs1 = F1*q1
+dobs2 = F2*q2
+
+@testset "FWI/LSRTM objective parallelization test with $(nlayer) layers and tti $(tti) and freesurface $(fs)" begin
+
+	ftol = 1f-5
+	
+	dm1 = 2f0*circshift(dm, 10)
+    dm2 = 2f0*circshift(dm, 30)
+
+	Jm0, grad = lsrtm_objective(model0, [q1, q2], [dobs1, dobs2], [dm1, dm2]; options=opt, nlind=true)
+    Jm01, grad1 = lsrtm_objective(model0, q1, dobs1, dm1; options=opt, nlind=true)
+    Jm02, grad2 = lsrtm_objective(model0, q2, dobs2, dm2; options=opt, nlind=true)
+    @test isapprox(Jm01, Jm0[1]; rtol=ftol)
```

Comment:
wait why multiple objective functions values not the sum?

Reply:
Sum over sources, not over vintages

Reply:
The objective value is a single value for the given problem. It's different multiple values if you have multiple independent LSRTM.

Reply:
OK agree. So the output should be 1 single objective, but multiple gradients.